cmd/agent/common/autoconfig.go

```diff
@@ -49,6 +49,7 @@ func SetupAutoConfig(confdPath string) {
 		filepath.Join(GetDistPath(), "conf.d"),
 	}
 	AC.AddProvider(providers.NewFileConfigProvider(confSearchPaths), false)
+	AC.AddProvider(providers.NewKubeletConfigProvider(), true)
```

Comment:
Should be added via `RegisterProvider` no? Especially because it's behind a build flag

Reply:
indeed, that was for testing. That'll be configured manually, and enabled automatically by the entrypoint